core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -188,6 +214,76 @@ public SimpleQueryBus(MessageMonitor<? super QueryMessage<?, ?>> messageMonitor,
                        .filter(Objects::nonNull);
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <Q, I, U> Registration subscriptionQuery(SubscriptionQueryMessage<Q, I, U> query,
+                                                    UpdateHandler<I, U> updateHandler) {
+        MessageMonitor.MonitorCallback monitorCallback = messageMonitor.onMessageIngested(query);
+        SubscriptionQueryMessage<Q, I, U> interceptedQuery = intercept(query);
+        List<QuerySubscription> subs = subscriptions
+                .computeIfAbsent(interceptedQuery.getQueryName(), k -> new CopyOnWriteArrayList<>())
+                .stream()
+                .filter(subscription -> interceptedQuery.getResponseType().matches(subscription.getResponseType()))
+                .collect(Collectors.toList());
+        Registration registration = () -> true;
+        try {
+            if (subs.isEmpty()) {
+                throw new NoHandlerForQueryException(
+                        format("No handler found for %s with response type %s and update type %s",
+                               interceptedQuery.getQueryName(),
+                               interceptedQuery.getResponseType(),
+                               interceptedQuery.getUpdateResponseType()));
+            }
+            Iterator<QuerySubscription> subsIterator = subs.iterator();
+            boolean invocationSuccess = false;
+            SimpleQueryUpdateEmitter emitter = new SimpleQueryUpdateEmitter(updateHandler);
+            while (!invocationSuccess && subsIterator.hasNext()) {
+                try {
+                    QuerySubscription subscription = subsIterator.next();
+                    if (subscription instanceof SubscribableQuerySubscription) {
+                        DefaultUnitOfWork<QueryMessage<Q, I>> uow = DefaultUnitOfWork.startAndGet(
```

Comment:
You can move this `uow` variable creation out side of the `if/else` block, as the call is identical  between line 244 and 262.

Reply:
Nice catch! This is a left over :)